tests/vmi_multus_test.go

```diff
@@ -375,32 +375,37 @@ var _ = Describe("[Serial]Multus", func() {
 		})
 
 		Context("VirtualMachineInstance with Linux bridge CNI plugin interface and custom MAC address.", func() {
-			interfaces := []v1.Interface{linuxBridgeInterface}
-			networks := []v1.Network{linuxBridgeNetwork}
-			linuxBridgeIfIdx := 0
 			customMacAddress := "50:00:00:00:90:0d"
 
 			It("[test_id:676]should configure valid custom MAC address on Linux bridge CNI interface.", func() {
 				By("Creating a VM with Linux bridge CNI network interface and default MAC address.")
-				vmiTwo := createVMIOnNode(interfaces, networks)
-				tests.WaitUntilVMIReady(vmiTwo, console.LoginToAlpine)
+				vmiTwo := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(linuxBridgeInterface),
+					libvmi.WithNetwork(&linuxBridgeNetwork),
+					libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
+					libvmi.WithCloudInitNoCloudNetworkData(cloudInitNetworkDataWithStaticIPsByDevice("eth1", "10.1.1.2/24"), false))
+				vmiTwo = tests.StartVmOnNode(vmiTwo, nodes.Items[0].Name)
 
 				By("Creating another VM with custom MAC address on its Linux bridge CNI interface.")
-				interfaces[linuxBridgeIfIdx].MacAddress = customMacAddress
-				vmiOne := createVMIOnNode(interfaces, networks)
-				tests.WaitUntilVMIReady(vmiOne, console.LoginToAlpine)
-
-				By("Configuring static IP address to the Linux bridge interface.")
-				Expect(configInterface(vmiOne, "eth0", "10.1.1.1/24")).To(Succeed())
-				Expect(configInterface(vmiTwo, "eth0", "10.1.1.2/24")).To(Succeed())
+				linuxBridgeInterfaceWithCustomMac := linuxBridgeInterface
+				linuxBridgeInterfaceWithCustomMac.MacAddress = customMacAddress
+				vmiOne := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(linuxBridgeInterfaceWithCustomMac),
+					libvmi.WithNetwork(&linuxBridgeNetwork),
+					libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
+					libvmi.WithCloudInitNoCloudNetworkData(cloudInitNetworkDataWithStaticIPsByMac(customMacAddress, "10.1.1.1/24"), false))
+				vmiOne = tests.StartVmOnNode(vmiOne, nodes.Items[0].Name)
+
+				tests.WaitUntilVMIReady(vmiOne, console.LoginToFedora)
+				tests.WaitUntilVMIReady(vmiTwo, console.LoginToFedora)
 
 				By("Verifying the desired custom MAC is the one that were actually configured on the interface.")
-				ipLinkShow := fmt.Sprintf("ip link show eth0 | grep -i \"%s\" | wc -l\n", customMacAddress)
-				err = console.SafeExpectBatch(vmiOne, []expect.Batcher{
-					&expect.BSnd{S: ipLinkShow},
-					&expect.BExp{R: "1"},
-				}, 15)
-				Expect(err).ToNot(HaveOccurred())
+				Expect(vmiOne.Status.Interfaces[1].MAC).To(Equal(customMacAddress), "should set linux bridge interface with the custom MAC address at VMI Status")
```

Comment:
1. Do you know for sure it will be on `vmiOne.Status.Interfaces[1]` ?
Order until now could be changed in VMI status
2. You need Eventually here, please see test 3985, else you might have the race condition between the GA and the VMI status.

Reply:
> Do you know for sure it will be on vmiOne.Status.Interfaces[1] ?
Order until now could be changed in VMI status

ack, will add a check

> You need Eventually here, please see test 3985 or your PR which fixes the race, else you might have the race condition between the GA and the VMI status.

The MAC is not taken from Guest Agent is taken from libvirt domxml so there is no need for Eventually.

Reply:
> > Do you know for sure it will be on vmiOne.Status.Interfaces[1] ?
> > Order until now could be changed in VMI status
> 
> ack, will add a check
> 
> > You need Eventually here, please see test 3985 or your PR which fixes the race, else you might have the race condition between the GA and the VMI status.
> 
> The MAC is not taken from Guest Agent is taken from libvirt domxml so there is no need for Eventually.

But the status might not be "ready" yet, since `WaitUntilVMIReady` just wait for AgentConnected,
which in turn just start the pollers, but it doesn't mean the status has already been updated.

Reply:
Mac is not taken from guest agent we don't need to wait for guest agent that's why I removed it.

Reply:
> Mac is not taken from guest agent we don't need to wait for guest agent that's why I removed it.

Mac is part of the reported part of guest agent for SRIOV for example, hence prone to the races,
and it should go all the way to VMI.status from what i understand.

```
{"return":[{"name":"lo","ip-addresses":[{"ip-address-type":"ipv4","ip-address":"127.0.0.1","prefix":8},{"ip-address-type":"ipv6","ip-address":"::1","prefix":128}],"statistics":{"tx-packets":0,"tx-errs":0,"rx-bytes":0,"rx-dropped":0,"rx-packets":0,"rx-errs":0,"tx-bytes":0,"tx-dropped":0},"hardware-address":"00:00:00:00:00:00"},{"name":"eth0","ip-addresses":[{"ip-address-type":"ipv4","ip-address":"10.0.2.2","prefix":24},{"ip-address-type":"ipv6","ip-address":"fe80::ff:feb4:ee86","prefix":64}],"statistics":{"tx-packets":13474,"tx-errs":0,"rx-bytes":123253841,"rx-dropped":0,"rx-packets":86812,"rx-errs":0,"tx-bytes":961261,"tx-dropped":0},"hardware-address":"02:00:00:b4:ee:86"},{"name":"eth1","hardware-address":"fe:00:a6:75:3b:b7"}]}
```

Reply:
This is not the case with non-sriov ifaces, so why we should do it here?

Reply:
Lost the code line, because it was updated since,
is it a secondary interface with bridge binding ?
if so it is also reported by GA i think ?
else you are right